src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -763,6 +822,57 @@ private void BeforeQueryStatusForPowerConsole(object sender, EventArgs args)
             }
         }
 
+        private void BeforeQueryStatusForUpgradeNuGetProject(object sender, EventArgs args)
+        {
+            ThreadHelper.JoinableTaskFactory.Run(async delegate
```

Comment:
Why are you using `ThreadHelper` instead of `NugetUIThreadHelper`? We need to make sure this task is joined correctly with the others because this will block the UI thread and could cause deadlocks.